js/core/tty/terminal.js

```diff
@@ -16,11 +16,13 @@
 const keyboard = require('../keyboard');
```

Comment:
I would not consider `tty` part of `core` because it's build on top of `keyboard` and `screen`, so move it outside. It could be considered a driver, although not a low-level real one (because it's build on top of other ones) but instead a "virtual" one.